src/clj/fluree/db/connection.cljc

```diff
@@ -1,31 +1,28 @@
 (ns fluree.db.connection
   (:require [clojure.core.async :as async]
+            [clojure.pprint :as pprint]
+            [clojure.string :as str]
             [fluree.db.constants :as const]
+            [fluree.db.storage :as storage]
             [fluree.db.util.core :as util :refer [get-first-value]]
             [fluree.db.util.log :as log :include-macros true]
             [fluree.db.util.async :refer [<? go-try]]
+            [fluree.db.serde.json :refer [json-serde]]
             [fluree.json-ld :as json-ld]
-            [fluree.db.ledger :as ledger]))
+            [fluree.db.ledger :as ledger])
+  #?(:clj (:import (java.io Writer))))
 
 #?(:clj (set! *warn-on-reflection* true))
 
 (defprotocol iConnection
-  (-close [conn] "Closes all resources for this connection")
-  (-closed? [conn] "Indicates if connection is open or closed")
   (-did [conn] "Returns optional default did map if set at connection level")
-  (-msg-in [conn msg] "Handler for incoming message from nameservices")
-  (-msg-out [conn msg] "Pushes outgoing messages/commands to connection service")
-  (-nameservices [conn] "Returns a sequence of all nameservices configured for the connection.")
-  (-state [conn] [conn ledger] "Returns internal state-machine information for connection, or specific ledger"))
+  (-nameservices [conn] "Returns a sequence of all nameservices configured for the connection."))
 
 (defprotocol iStorage
   (-c-read [conn commit-key] "Reads a commit from storage")
   (-c-write [conn ledger-alias commit-data] "Writes a commit to storage")
   (-txn-write [conn ledger-alias txn-data] "Writes a transaction to storage and returns the key. Expects string keys.")
-  (-txn-read [conn txn-key] "Reads a transaction from storage")
-  (-index-file-write [conn ledger-alias idx-type index-data] "Writes an index item to storage")
-  (-index-file-read [conn file-address] "Reads an index item from storage")
-  (-index-file-delete [conn file-address] "Deletes an index item from storage"))
+  (-txn-read [conn txn-key] "Reads a transaction from storage"))
 
 (comment
```

Comment:
Is this comment still accurate? My understanding is that the Connection ledger state is a map of ledger-alias->channel[Ledger].